src/MerchantCenter/PolicyComplianceCheck.php

```diff
@@ -68,6 +70,116 @@ public function is_accessible(): bool {
 		return true;
 	}
 
+	/**
+	 * Check if the store sample product landing pages lead to a 404 error.
+	 *
+	 * @return bool
+	 */
+	public function has_page_not_found_error(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headers ) && $headers[0] !== 'HTTP/1.1 200 OK' ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Check if the store sample product landing pages has redirects through 3P domains.
+	 *
+	 * @return bool
+	 */
+	public function has_redirects(): bool {
+		$headers = $this->get_landing_page_headers();
+		if ( ! empty( $headesr ) && isset( $headers[0] ) ) {
+			if ( $headers[0] === 'HTTP/1.1 302 Found' ) {
+				// this is the URL where it's redirecting
+				return true;
+			}
+		}
+			return false;
```

Comment:
As mentioned in the previous comment it's better to use `wp_remote_retrieve_response_code` to retrieve the response code. Also the string comparison isn't very accurate because there are [many redirect codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_redirection). Lastly by default a `wp_remote_get` will follow redirects so we need to set the `redirection` argument to 0 to detect a redirect response code.

So I suggest to create this function like this:
```suggestion
		$url      = $this->get_landing_page_url();
		$response = wp_remote_get( $url, [ 'redirection' => 0 ] );
		$code     = wp_remote_retrieve_response_code( $response );

		if ( $code >= 300 && $code <= 399 ) {
			return true;
		}

		return false;
```